src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponentResult.cs

```diff
@@ -0,0 +1,125 @@
+        /// </summary>
+        public MediaTypeHeaderValue ContentType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the HTTP status code.
+        /// </summary>
+        public int? StatusCode { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="ITempDataDictionary"/> for this result.
+        /// </summary>
+        public ITempDataDictionary TempData { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the view component to invoke. Will be ignored if <see cref="ViewComponentType"/>
+        /// is set to a non-<c>null</c> value.
+        /// </summary>
+        public string ViewComponentName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the type of the view component to invoke.
+        /// </summary>
+        public Type ViewComponentType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="ViewDataDictionary"/> for this result.
+        /// </summary>
+        public ViewDataDictionary ViewData { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="IViewEngine"/> used to locate views.
+        /// </summary>
+        /// <remarks>When <c>null</c>, an instance of <see cref="ICompositeViewEngine"/> from
+        /// <c>ActionContext.HttpContext.RequestServices</c> is used.</remarks>
+        public IViewEngine ViewEngine { get; set; }
+
+        /// <inheritdoc />
+        public override async Task ExecuteResultAsync(ActionContext context)
+        {
+            var response = context.HttpContext.Response;
+            var services = context.HttpContext.RequestServices;
+
+            var htmlHelperOptions = services.GetRequiredService<IOptions<MvcViewOptions>>().Options.HtmlHelperOptions;
+            var viewComponentHelper = services.GetRequiredService<IViewComponentHelper>();
+
+            var viewData = ViewData;
+            if (viewData == null)
+            {
+                var modelMetadataProvider = services.GetRequiredService<IModelMetadataProvider>();
+                viewData = new ViewDataDictionary(modelMetadataProvider, context.ModelState);
+            }
+
+            var contentType = ContentType ?? ViewExecutor.DefaultContentType;
+            if (contentType.Encoding == null)
+            {
+                // Do not modify the user supplied content type, so copy it instead
+                contentType = contentType.Copy();
+                contentType.Encoding = Encoding.UTF8;
+            }
+
+            if (StatusCode != null)
+            {
+                response.StatusCode = (int)StatusCode;
+            }
+
+            response.ContentType = contentType.ToString();
+
+            using (var writer = new HttpResponseStreamWriter(response.Body, contentType.Encoding))
+            {
+                var viewContext = new ViewContext(
+                    context,
+                    NullView.Instance,
```

Comment:
Thoughts on null-object pattern here? I don't like the idea of allowing the view argument to be null.